lib/global_id/identification.rb

```diff
@@ -21,5 +21,31 @@ def to_signed_global_id(options = {})
     def to_sgid_param(options = {})
       to_signed_global_id(options).to_param
     end
+
+    ActiveRecord::Associations::Builder::BelongsTo.instance_eval do
+      def define_accessors(model, reflection)
+        class_name = reflection.options.fetch(:class_name, reflection.name.capitalize.to_s)
```

Comment:
Ideally this will call super but to be able to call super this should be extracted in a module and the module mixed in with .prepend